int32 machine_num
MachineTask[] machine_tasks